io-sim-classes/src/Control/Monad/Class/MonadSTM/Strict.hs

```diff
@@ -113,9 +127,14 @@ newTVarWithInvariantIO :: (MonadSTM m, HasCallStack)
                        => (a -> Maybe String) -- ^ Invariant (expect 'Nothing')
                        -> a
                        -> m (StrictTVar m a)
-newTVarWithInvariantIO invariant !a =
+#if CHECK_TVAR_INVARIANT
```

Comment:
This seems unnecessary, since the definition of `checkInvariant` already uses CPP to make it a no op without the flag.

Reply:
Ah, nevermind. I misread the difference. It's about `invariant` being in scope.